Update Build Instructions

davschneller opened the pull request:
A bit overdue, but we should be able to update our build instructions a bit. So this PR shall do so.

It's still WIP, especially the `Compiling SeisSol` section. Have we documented the CMake parameters anywhere so far? Or will someone need to do that still?

sebwolf-de commented on Documentation/compiling-seissol.rst:
```diff
@@ -1,44 +1,96 @@
 Compiling SeisSol
 ~~~~~~~~~~~~~~~~~
-Get the latest version of SeisSol on git by cloning the whole repository
+Get the latest version of SeisSol on git by cloning the whole repository,
```
Should we add something like: "Get the latest release (https://github.com/SeisSol/SeisSol/releases), by cloning the whole repository: `git clone --recursive --branch vX.Y.Z https://github.com/SeisSol/SeisSol.git`"

davschneller replied:
I'll add it. Maybe also a note that the master branch may be unstable? (but contains the latest features)

sebwolf-de replied:
Yes, that's a good idea.

sebwolf-de commented on Documentation/compiling-seissol.rst:
```diff
@@ -1,44 +1,96 @@
-   mkdir build-release && cd build-release
-   CC=mpicc CXX=mpiCC FC=mpif90 cmake -DCOMMTHREAD=ON -DNUMA_AWARE_PINNING=ON -DASAGI=ON -DCMAKE_BUILD_TYPE=Release -DHOST_ARCH=skx -DPRECISION=double -DORDER=4 -DGEMM_TOOLS_LIST=LIBXSMM,PSpaMM ..
+   mkdir build && cd build
+   cmake -DCMAKE_BUILD_TYPE=Release -DHOST_ARCH=hsw -DPRECISION=double -DORDER=4 -DGEMM_TOOLS_LIST=Eigen -DEQUATIONS=elastic -DGRAPH_PARTITIONING_LIBS=none ..
```
Given the fact that a fair amount of people will just copy-paste this line, should we use `hsw`, `libxsmm`, `parmetis` as default?

sebwolf-de replied:
Also, why not `GEMM_TOOLS_LIST=auto`?

sebwolf-de replied:
I'd vote for simplicity, which is to provide one set of instructions, which works for almost everybody with good performance. 
Installing libxsmm is easy, every user should do it sooner or later anyways. If they've compiled netcdf/hdf5 successfully, they will also easily install libxsmm. Thus, I don't see a reason for distinguishing between minimal/optimal, but I would just give a default.

sebwolf-de commented on Documentation/compiling-seissol.rst:
```diff
@@ -1,44 +1,96 @@
+The following parameters are important to note:
+
+``PRECISION`` is either ``single`` or ``double`` and corresponds to 32-bit floating point numbers and 64-bit floating point numbers
+used during all calculations. If your simulation gives an error when using single precision, it can sometimes be fixed by re-compiling for double precision.
```
`If you experience Inf/Nan errors, when using single precision...`

sebwolf-de commented on Documentation/compiling-seissol.rst:
```diff
@@ -1,44 +1,96 @@
+The ``ORDER`` parameter may range from 2 to 8. The ``EQUATIONS`` parameter may be ``elastic``, ``viscoelastic2``, ``poroelastic``,
+or ``anisotropic``. For the ``viscoelastic2`` setting, you will need to add e.g. ``-DNUMBER_OF_MECHANISMS=3`` (or replace 3 by any other number greater than zero).
+
+Once you have installed a GEMM generator, choose a different option in ``GEMM_TOOLS_LIST`` to obtain a faster version of SeisSol.
```
Would change this to:
```
The GEMM generator is used to create optimized code for matrix-matrix multiplications. 
* `libxsmm` should give reasonable performance on most x86 CPUs.
* For the `skx` architecture, we suggest `libxsmm,PSPaMM`.
* If you have trouble with these libraries, you can use `Eigen`, which should work on most available architectures.

sebwolf-de commented on Documentation/compiling-seissol.rst:
```diff
@@ -1,44 +1,96 @@
+     - IBM PowerPC 9
+     - 
+
+In case of a manual installation of dependencies (and not using the `setup.sh` script from the "Installing Dependencies" page), you may have to prepend :code:`CMAKE_PREFIX_PATH` and :code:`PKG_CONFIG_PATH` to the cmake command, e.g. for dependencies installed in :code:`${PREFIX}`:
```
It is important that all dependencies and SeisSol are compiled with the same compiler. In particular, `gcc` and `intel` compilers are not compatible.
On some clusters, where several compilers are installed, you might want to specifically point to the compiler, which you want to use for SeisSol. For the `intel` compiler use e.g. `CC=mpiicc CXX=mpiicpc cmake ...`.

krenzland replied:
gcc/clang however are *typically* not a problem

sebwolf-de replied:
Maybe: `In particular, we have experienced problems when mixing gcc/clang with intel compilers`.

davschneller replied:
We don't particularly need to use `mpicc`/`mpiicc` (and `mpicxx`/`mpiicpc`), do we? I.e. the non-MPI compilers should be good... Or are they?

sebwolf-de replied:
In my experience, using the non-mpi compilers, typically cmake can't find the proper libraries. That's why I always use the mpi wrappers.

Thomas-Ulrich replied:
I've tested the non-mpi compilers (on a gcc12 openmpi compilation) and it was working fine.

sebwolf-de commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
 
+All dependencies can be installed automatically with spack or manually by hand.
+For a minimal installation (e.g. for running SeisSol on your local machine),
+you may avoid all optional dependencies. However, for the maximal performance, and for compute clusters,
```
I would split it:
```
For maximal node-level performance, we recommend using a code generator.
On distributed-memory computers, a mesh partitioner reduces the load-imbalance.
Using the libraries installed by the cluster admins (modules), might give a little performance boost, e.g. when using optimized I/O routines.
```

sebwolf-de commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
-Initial Adjustments to .bashrc
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+We recommend setting up a folder on your system which will contain the build files for all dependencies.
+For example, do `mkdir ~/seissol; cd seissol`.
```
Do you really think `seissol` is a good name? It will clash with the name of the git repo. I'd use `~/seissol_dependencies`

sebwolf-de commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
+  export CPATH=$PREFIX/include:$CPATH 
+  export C_INCLUDE_PATH=$PREFIX/include:$C_INCLUDE_PATH
+  export CXX_INCLUDE_PATH=$PREFIX/include:$CXX_INCLUDE_PATH
+  export EDITOR=nano # or e.g. vi,vim
```
nano?! But let's not start an editor war here ;-)

davschneller replied:
It's the easiest-to-use. Otherwise, I agree—I'd prefer to write `vim` in there anyways... :D

sebwolf-de commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
 .. code-block:: bash
 
   # you will need at least version 3.20.0 for GNU Compiler Collection 
   (cd $(mktemp -d) && wget -qO- https://github.com/Kitware/CMake/releases/download/v3.20.0/cmake-3.20.0-Linux-x86_64.tar.gz | tar -xvz -C "." && mv "./cmake-3.20.0-linux-x86_64" "${HOME}/bin/cmake")
```
It's `${HOME}/bin/make` here, but `${PREFIX}/bin/...` below

davschneller replied:
Thanks! Missed that one. Maybe `PREFIX` should be `SEISSOL_PREFIX` instead, that would make it more easy to recognize. The choice for `PREFIX` was mainly since if some `make` script has a prefix variable, it's sometimes named `PREFIX`, i.e. setting the environment variable in advance will spare us from setting that explicitly again in this case. ... But we actually don't really run into that, so... I'll rename it.

sebwolf-de commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
-~~~~~~~~~~~~~~~
+"""""""""""""""
+
+We begin with HDF5 which is needed for reading meshes in PUML format (the default format in SeisSol, and the output of PUMgen).
```
`HDF5, which`

Thomas-Ulrich replied:
I would remove "We begin with".
HDF5 is needed ...

sebwolf-de commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
+- PSpaMM (pspamm.py) for small sparse matrix multiplications (required only on Knights Landing or Skylake)
+
+Note that using Eigen does not result in any additional dependencies, since it is needed in SeisSol anyways.
+Other than that, we recommend using the combination libxsmm and PSpaMM.
```
We recommend `libxsmm` and in addition `PSpaMM` for architectures with `AVX-512`.

sebwolf-de commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
+  wget https://syncandshare.lrz.de/dl/fiJNAokgbe2vNU66Ru17DAjT/netcdf-4.6.1.tar.gz
+  tar -xaf netcdf-4.6.1.tar.gz
+  cd netcdf-4.6.1
+  CFLAGS="-fPIC ${CFLAGS}" CC=h5pcc ./configure --enable-shared=no --prefix=$HOME --disable-dap
```
`prefix=$PREFIX`?!

davschneller replied:
See the comment above, it's actually correct here.

sebwolf-de commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
-Installing SYCL (for GPUs)
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+ASAGI
+"""""
 
-See section :ref:`Installing SYCL <installing_SYCL>`.
+See section :ref:`Installing ASAGI <installing_ASAGI>`.
```
Add: `If you want to use SeisSol with ASAGI, you need to install `easi` with ASAGI support.`

sebwolf-de reviewed (changes requested):
Good idea to refactor this, some ideas from my side.
IMHO, we have to be careful with `$PREFIX`. The most important part is that it is consistent everywhere.

sebwolf-de commented:
Maybe a general remark: I'd add a remark: "This documentation is tested for [Ubuntu 22.04 LTS]." Then we should be sure that the documentation actually works in a fresh installed Ubuntu 22.04 (maybe test with docker).

Thomas-Ulrich commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
   tar -xaf hdf5-1.10.8.tar.bz2
   cd hdf5-1.10.8
-  CPPFLAGS="-fPIC ${CPPFLAGS}" CC=mpicc FC=mpif90 ./configure --enable-parallel --prefix=$HOME --with-zlib --disable-shared --enable-fortran 
+  CPPFLAGS="-fPIC ${CPPFLAGS}" CC=mpicc CXX=mpicxx ./configure --enable-parallel --prefix=$PREFIX --with-zlib --disable-shared
```
the problem is that the compilers may have other names.

Thomas-Ulrich commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
-
-Installing Eigen3
-~~~~~~~~~~~~~~~~~
+Next, we look at Eigen which conveniently uses CMake as a build system for itself.
```
I would remove this sentence.

Thomas-Ulrich commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
    cd ../..
 
-.. _installing_libxsmm:
+Lastly, we need easi which is (most likely) not already installed on your system or as a module file, as it is a more SeisSol-specific library.
```
Easi is used for setting up the model parameters.
It is (most likely) not already installed on your system or as a module file, as it is a more SeisSol-specific library.

Thomas-Ulrich commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
+It is used for setting up the model parameters.
+Here you can find the `installation instructions <https://easyinit.readthedocs.io/en/latest/getting_started.html>`_.
+
+And with that, we're good to go!
```
I would remove this one.

Thomas-Ulrich commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
+Code Generators for CPUs (optional, recommended)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For CPU code generators, we support the following:
```
we support the following CPU code generators

Thomas-Ulrich commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
+- libxsmm (libxsmm\_gemm\_generator) for small matrix multiplications
+- PSpaMM (pspamm.py) for small sparse matrix multiplications (required only on Knights Landing or Skylake)
+
+Note that using Eigen does not result in any additional dependencies, since it is needed in SeisSol anyways.
```
I would remove this sentence, and use Sebastian sentence.

Thomas-Ulrich commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
+
+Once you have SYCL and GemmForge (maybe also ChainForge) ready, you are set for compiling SeisSol with GPUs.
+
+Code Generators for CPUs (optional, recommended)
```
remove for CPUs
(you talk about GPU code generators in this section).

Thomas-Ulrich commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
-~~~~~~~~~~~~~~~~~~
+""""""""""""""""""
+
+(to save data, we only use a shallow clone)
```
I would remove this sentence.

Thomas-Ulrich commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
-
+"""""""""""""""""
 
+(to save data, we only use a shallow clone)
```
remove

Thomas-Ulrich reviewed (changes requested):
thank you for these changes.

davschneller commented on Documentation/installing-dependencies.rst:
```diff
@@ -1,216 +1,293 @@
 
-   pip3 install --user git+https://github.com/SeisSol/gemmforge.git
-   pip3 install --user git+https://github.com/SeisSol/chainforge.git
+  wget https://syncandshare.lrz.de/dl/fiJNAokgbe2vNU66Ru17DAjT/netcdf-4.6.1.tar.gz
```
That was there from before. :) And no one changed it.

vikaskurapati commented on Documentation/build-archs.rst:
```diff
@@ -0,0 +1,254 @@
+     -
+     - 
+   * - ``apple-m2``
+     - Apple M1 CPU
```
Apple M2

vikaskurapati commented on Documentation/build-overview.rst:
```diff
@@ -0,0 +1,16 @@
+Building SeisSol
```
Can we change this to "Installing SeisSol"? Not something big but staying consistent with the line below would be nicer, IMO...

Thomas-Ulrich replied:
+1

vikaskurapati commented on Documentation/build-overview.rst:
```diff
@@ -0,0 +1,16 @@
+* Then, compile SeisSol. See the instructions HERE.
+
+
+Once you are done, you can test your installation by running some examples.
```
Can we have instructions to run proxy as another testing possibility?

vikaskurapati commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,98 @@
+The following parameters will alter the name of the SeisSol executable.
+You may explicitly compile and install multiple of these configurations at the same time—however, you will have to re-do the CMake and build process for each configuration.
+
+- ``CMAKE_BUILD_TYPE``: Either
```
Remove Either? As it is not in line with the description given for the other parameters..

vikaskurapati commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,98 @@
+(i.e. add a "c" in front of the command)
+Alternatively, you may also modify the ``CMakeCache.txt`` in your build directory directly (for it to exist, ``cmake`` needs to be run first).
+
+Equation-specific parameters
```
Can we move it below CMAKE_BUILD_TYPE? As the build type is not really an equation-specific parameter.

vikaskurapati commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,98 @@
+- ``EQUATION``: the equation system to compile for
+
+    * ``elastic``: isotropic elastic
+    * ``viscoelastic``: obsolete, use ``viscoelastic2`` instead
```
Can we eventually get rid of this if this is obsolete?

vikaskurapati commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,98 @@
+    * ``anisotropic``: anisotropic elastic; essentially uses the same kernels as the ``elastic`` mode, but with more general matrices
+    * ``poroelastic``: isotropic poroelastic
+- ``NUMBER_OF_MECHANISMS``: the number of mechanisms for viscoelastic simulations. For all other equations, this parameter is required to be 0.
+- ``ORDER``: the order of the used polynomial used, that is, its degree plus 1. The order is used for both space and time integration. For example, if you specify order 4, you will be using polynomials of degree 3 in space and time. Note that a higher order can impact the performance greatly.
```
All these aren't equation specific parameters, either. Can we push down into Generic parameters or have a separate section for them?

davschneller commented:
Hi everyone; sorry for the huge delays with handling this PR.

Most of the issues should be addressed by now; mind looking over it once more? (also, I've added some additional content regarding running SeisSol; and added licensing info everywhere)

Thomas-Ulrich commented on Documentation/gpus.rst:
```diff
@@ -8,29 +13,21 @@ General
 .. _gpu_process_pinning:
 
 The current GPU version of SeisSol targets the latest NVidia Graphics cards. Therefore, you
```
and AMD

Thomas-Ulrich commented on Documentation/gpus.rst:
```diff
@@ -8,29 +13,21 @@ General
+need to have at least **CUDA 11** or ROCm 5.2 installed in your environment. Moreover, make sure
+that your installed MPI implementation is GPU-aware. If you are unsure, consult the documentation of the cluster you are using.
+
+You can also `UCX communication layer
```
verb missing?

Thomas-Ulrich commented on Documentation/gpus.rst:
```diff
@@ -126,25 +124,23 @@ the following will force SeisSol to allocate 1.5GB of stack GPU memory for tempo
-To enable it, set "SEISSOL_SERIAL_NODE_DEVICE_INIT=1". To explicitly disable it,
-write "SEISSOL_SERIAL_NODE_DEVICE_INIT=0".
+To enable it, set ``SEISSOL_SERIAL_NODE_DEVICE_INIT=1``. To explicitly disable it,
+write ``SEISSOL_SERIAL_NODE_DEVICE_INIT=0``.
```
maybe with export?

Thomas-Ulrich commented on Documentation/build-archs.rst:
```diff
@@ -0,0 +1,259 @@
+     - Generates plain x86-64 instructions, without SIMD instructions like SSE/AVX/AMX etc.
+     -
+     -
+   * - ``wsm``
```
should this table have a limit in cell width?
note that it is extremly wide.

Thomas-Ulrich commented on Documentation/build-archs.rst:
```diff
@@ -0,0 +1,259 @@
+     - ``hip``
+     - AMD GCN 5 (Vega)
+     - AMD Instinct MI25, Radeon RX Vega 56, Radeon RX Vega 64
+     - splitFOOTNOTE1
```
splitFOOTNOTE1 does not work

Thomas-Ulrich commented on Documentation/build-dependencies.rst:
```diff
@@ -0,0 +1,321 @@
+  - Clang (tested: 18, 19)
+  - NVHPC (tested: 24.09; currently still slow!)
+  - Cray CE (however: no commthread support; needs ``SEISSOL_COMMTHREAD=0``)
+  - ICC 2021.9 is supported up to v1.2.0 only
```
now works

davschneller replied:
You mean, it works now again?

Thomas-Ulrich replied:
yes (you fixed it with https://github.com/SeisSol/SeisSol/pull/1210)

Thomas-Ulrich commented on Documentation/build-dependencies.rst:
```diff
@@ -0,0 +1,321 @@
+Installation with Spack
+-----------------------
+
+The `Spack <https://github.com/spack/spack/wiki>`_ package ``seissol-env`` allows to automatically install all dependencies of SeisSol (e.g. mpi, hdf5, netcdf, easi, asagi, etc),
```
shouldn't we better describe the seissol package?
(seissol-env package is only maintained because of the outdated spack on supermuc. they also have a new spack stack, but their tuning of the modules makes it difficult using it only for loading the dependencies, for latter manually compilation).

Thomas-Ulrich replied:
some good but outdated usage of the spack package are heisenbug or frontera documentations.

davschneller replied:
Ok, done.

Thomas-Ulrich commented on Documentation/build-seissol.rst:
```diff
@@ -0,0 +1,190 @@
+-----------------------------
+
+For this page, we will assume that all the environment variables
+from the previous section to be loaded (cf. section ). That is, you have your SeisSol build directory in ``$SEISSOL_BASE``.
```
ref missing (cf. section)

Thomas-Ulrich commented on Documentation/build-seissol.rst:
```diff
@@ -0,0 +1,190 @@
+Sometimes, the ninja build tool can speed up your build process. To use it, add ``-GNinja`` to the cmake command (you might need to delete your cmake folder first to switch from make).
+
+Generally, we recommend using ``PRECISION=single`` for faster simulations. If you experience Inf/NaN errors (cf. also https://github.com/SeisSol/SeisSol/issues/200 ), then setting ``PRECISION=double`` is a good idea.
+The ``ORDER`` equals the degree of the used polynomials plus one. In general, we optimize for the orders 3 to 6 the most.
```
"In general, we optimize for the orders 3 to 6 the most"
why?
Maybe remove statement

Thomas-Ulrich commented on Documentation/build-seissol.rst:
```diff
@@ -0,0 +1,190 @@
+Currently, SeisSol builds have the following constraints: they are restricted to one PDE, one precision and one polynomial degree usage for discretization
+
+* a single equation system (isotropic elastic, anisotropic elastic, viscoelastic, poroelastic)
+* a single polynomial discretization degree (2 to 7)
```
8?

Thomas-Ulrich commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,105 @@
+Build Parameter Reference
+=========================
+
+**For Developers:** if you add a CMake parameter, make sure to document it here.
```
I would remove this sentence. it seems clear that if you add new features you should document them.

davschneller replied:
I'd actually see it differently here: it can never be clear enough. :)
Unless it's placed better somewhere else, of course.

Thomas-Ulrich commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,105 @@
+(i.e. add a "c" in front of the command)
+Alternatively, you may also modify the ``CMakeCache.txt`` in your build directory directly (for it to exist, ``cmake`` needs to be run first).
+
+Simulation- and Optimization-specific Parameters
```
remove capital letters

Thomas-Ulrich commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,105 @@
+    * ``anisotropic``: anisotropic elastic; essentially uses the same kernels as the ``elastic`` mode, but with more general matrices
+    * ``poroelastic``: isotropic poroelastic
+- ``NUMBER_OF_MECHANISMS``: the number of mechanisms for viscoelastic simulations. For all other equations, this parameter is required to be 0.
+- ``ORDER``: the order of the used polynomial used, that is, its degree plus 1. The order is used for both space and time integration. For example, if you specify order 4, you will be using polynomials of degree 3 in space and time. Note that a higher order can impact the performance greatly.
```
the expected convergence order

Thomas-Ulrich commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,105 @@
+
+- ``ASAGI``: enable or disable ASAGI as input for fault and material information. Note that easi will need to be built with ASAGI support if you select this option.
+- ``COVERAGE``: determine the code coverage (currently only relevant for the CI).
+- ``DR_QUAD_RULE``: the quadrature rule used for the 
```
unfinished sentence

Thomas-Ulrich commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,105 @@
+The following options are available, but need to be left in the state that they are in. Not doing so will most likely break the build process or the software.
+
+- ``INTEGRATE_QUANTITIES``: assumed to be always disabled. Currently broken; it will probably be replaced in some version soon—when we refactor the IO component of SeisSol.
+- ``MPI``: assumed to be always enabled. Builds without MPI will most likely give incorrect results at the moment. If you want to build without MPI support, you will at the current moment have to explicitly disable the graph partitioning, as each of the supported libraries will need MPI on its own already. Official support for builds without MPI is very low priority.
```
it will not build

Thomas-Ulrich commented on Documentation/build-parameters.rst:
```diff
@@ -0,0 +1,105 @@
+
+- ``INTEGRATE_QUANTITIES``: assumed to be always disabled. Currently broken; it will probably be replaced in some version soon—when we refactor the IO component of SeisSol.
+- ``MPI``: assumed to be always enabled. Builds without MPI will most likely give incorrect results at the moment. If you want to build without MPI support, you will at the current moment have to explicitly disable the graph partitioning, as each of the supported libraries will need MPI on its own already. Official support for builds without MPI is very low priority.
+- ``OPENMP``: assumed to be always enabled. Probably more dispensable than the ``MPI`` flag, but no guarantee (nor testing) for correctness is given. There are currently no specific plans to make this
```
unfinished sentence.

Thomas-Ulrich reviewed (changes requested):
Thank you for this effort. See comments attached

davschneller commented on Documentation/memory-requirements.rst:
```diff
@@ -1,3 +1,8 @@
+..
+  SPDX-FileCopyrightText: 2020-2024 SeisSol Group
```
Hopefully not; it's only now mainly for the copyright additions and rework of some of the content; for everything else then only if someone changes it. :D 

(alternatively, some also use the style of printing only the beginning year)

Thomas-Ulrich reviewed (approved):
I've pushed a commit to do some small things that you forget to address, maybe you want to have a look before merging.
switched back to netcdf-c 4.8.1 because of the recent asagi issue

davschneller commented:
Thanks a bunch; looking good to me.
